fog/ledger/server/src/key_image_service.rs

```diff
@@ -97,12 +97,16 @@ impl<L: Ledger + Clone, E: LedgerEnclaveProxy> KeyImageService<L, E> {
 
     // Helper function that is common
     fn enclave_err_to_rpc_status(&self, context: &str, src: EnclaveError) -> RpcStatus {
-        // Treat prost-decode error as an invalid arg, everything else is an internal
-        // error
+        // Treat prost-decode error as an invalid arg,
+        // treat attest error as permission denied,
+        // everything else is an internal error
         match src {
             EnclaveError::ProstDecode => {
                 rpc_invalid_arg_error(context, "Prost decode failed", &self.logger)
             }
+            EnclaveError::Attest(err) => {
```

Comment:
This pattern exists in other places in the code (e.g. https://github.com/mobilecoinfoundation/mobilecoin/blob/5302c33c347be0d50d2a525f76142aa206a8d8f2/fog/view/server/src/fog_view_service.rs#L117) but was missing here.

Reply:
Thank you! This is bug is what 1.1.1 was supposed to fix in fog-view side.
I thought we were doing this correctly everywhere in fog-ledger but apparently not?
It looks like the merkle proof service is correctly doing it

Reply:
Apparently not...